File: allSpice/Models/Recipe.cs
namespace allSpice.Models;

public class Recipe : ICreated, IRepoItem<int>
{
  public int Id { get ; set ; }
  public DateTime CreatedAt { get ; set ; }
  public DateTime UpdatedAt { get ; set ; }
  public string CreatorId { get ; set ; }
  public Profile Creator { get ; set ; }

  public string Title { get; set; }
  public string Instructions { get; set; }
  public string Img { get; set; }
  public string Category { get; set; }
  public  int FavoriteCount {get; set;}
  public int IngredientCount{get; set;}
  public bool favorited {get; set;}
}

File: allSpice.client/src/models/Recipe.js
import { AppState } from "../AppState.js";
import { Account } from "./Account.js";

export class Recipe {
  constructor(data) {
    this.id = data.id;
    this.creatorId = data.creatorId;
    this.creator = new Account(data.creator);
    this.createdAt = data.createdAt;
    this.updatedAt = data.updatedAt;
    this.title = data.title;
    this.instructions = data.instructions;
    this.img = data.img;
    this.category = data.category;
    this.favoriteCount = data.favoriteCount;
    this.archived = data.archived;
    this.favorited = getFavorite(data.id);
  }
}

function getFavorite(id) {
  let fav = AppState.favorites.find((f) => f.id == id);
  if (fav) {
    return true;
  }
  return false;
}
